code/game/objects/effects/spiders.dm

```diff
@@ -5,46 +5,27 @@
 	desc = "it's stringy and sticky"
 	anchored = 1
 	density = 0
-	var/health = 15
+	obj_integrity = 15
 
-//similar to weeds, but only barfed out by nurses manually
-/obj/structure/spider/ex_act(severity, target)
-	switch(severity)
-		if(1)
-			qdel(src)
-		if(2)
-			if (prob(50))
-				qdel(src)
-		if(3)
-			if (prob(5))
-				qdel(src)
-
-/obj/structure/spider/attacked_by(obj/item/I, mob/user)
-	..()
-	var/damage = I.force
-	take_damage(damage, I.damtype, 1)
-
-/obj/structure/spider/bullet_act(obj/item/projectile/P)
+
+
+/obj/structure/spider/play_attack_sound(damage_amount, damage_type = BRUTE, damage_flag = 0)
+	if(damage_type == BURN)//the stickiness of the web mutes all attack sounds except fire damage type
+		playsound(loc, 'sound/items/Welder.ogg', 100, 1)
+
+
+/obj/structure/spider/run_obj_armor(damage_amount, damage_type, damage_flag = 0, attack_dir)
+	if(damage_flag == "melee")
+		switch(damage_type)
+			if(BURN)
+				damage_amount *= 2
+			if(BRUTE)
+				damage_amount *= 0.25
 	. = ..()
-	take_damage(P.damage, P.damage_type, 0)
-
-/obj/structure/spider/proc/take_damage(damage, damage_type = BRUTE, sound_effect = 1)
-	switch(damage_type)
-		if(BURN)
-			damage *= 2
-			if(sound_effect)
-				playsound(loc, 'sound/items/Welder.ogg', 100, 1)
-		if(BRUTE)//the stickiness of the web mutes all attack sounds except fire damage type
-			damage *= 0.25
-		else
-			return
-	health -= damage
-	if(health <= 0)
-		qdel(src)
 
 /obj/structure/spider/temperature_expose(datum/gas_mixture/air, exposed_temperature, exposed_volume)
 	if(exposed_temperature > 300)
-		take_damage(5, BURN, 0)
+		take_damage(5, BURN, 0, 0)

```

Comment:
damage coming from heat get not specific damage flag, we can't have it use "fire" because some objects are immune to fire but should still take damage from excessive heat.